File: amazia-core/src/main/java/com/example/user/controller/CreateUserController.java
package com.example.user.controller;

import com.example.user.dto.CreateUserRequest;
import com.example.user.dto.UserResponse;
import com.example.user.service.CreateUserService;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/users")
public class CreateUserController {

    private final CreateUserService createUserService;

    public CreateUserController(CreateUserService createUserService) {
        this.createUserService = createUserService;
    }

    @PostMapping
    public ResponseEntity<UserResponse> create(@Valid @RequestBody CreateUserRequest request) {
        return ResponseEntity.status(HttpStatus.CREATED).body(createUserService.create(request));
    }
}

File: amazia-console/app/User/Controller/CreateUserController.php
<?php

namespace App\User\Controller;

use App\Http\Controllers\Controller;
use App\User\Service\CreateUserService;
use Illuminate\Http\Request;

class CreateUserController extends Controller
{
    private CreateUserService $service;

    public function __construct(CreateUserService $service)
    {
        $this->service = $service;
    }

    public function __invoke(Request $request)
    {
        $response = $this->service->create($request->all());
        return response()->json($response->json(), $response->status());
    }
}
